lib/component-data/pubsub.js

```diff
@@ -287,15 +290,20 @@ function updateComponentInstances({ eventID, snapshot, pubData, store }) {
       const instances = _.filter(Object.keys(store.state.components), (uri) => _.includes(uri, `${componentRoute}${name}/`) && !_.isEmpty(store.state.components[uri]));
 
       return Promise.all(_.compact(_.map(instances, (uri) => {
+        // the saveComponent dispatch is wrapped in a $nextTick to prevent a race condition where one component is subscribed to multiple values, which can result in each subsequent save overwriting the values in the previous ones
         if (scope && subscriberInScope({ scope, publisherURI, subscriberURI: uri, pubData, store })) {
           // subscriber is in scope!
-          return store.dispatch('saveComponent', { uri, data, eventID, snapshot });
+          return vue.$nextTick().then(()=> {
+            return store.dispatch('saveComponent', { uri, data, eventID, snapshot });
+          });
         } else if (scope) {
           // not in scope!
           return;
         } else {
           // no scope!
-          return store.dispatch('saveComponent', { uri, data, eventID, snapshot });
+          return vue.$nextTick().then(()=> {
```

Comment:
```suggestion
          return vue.$nextTick().then(() => {
```

Reply:
A good example of what a style guide/linter would help with

Reply:
👍